codalab/worker/worker.py

```diff
@@ -239,12 +217,50 @@ def cleanup(self):
         logger.info("Stopped Worker. Exiting")
 
     def signal(self):
-        # When the terminate flag is False, set the stop flag to stop running
-        # the worker without changing the status of existing running bundles.
-        if not self.terminate:
+        """
+        When the pass_down_termination flag is False, set the stop flag to stop running
+        the worker without changing the status of existing running bundles. Otherwise,
+        restage all bundles that are not in the terminal states [FINISHED, RESTAGED].
```

Comment:
more accurate to say "set a flag to restage..."

Reply:
Updated.